src/Controllers/CopyPatrol.php

```diff
@@ -123,6 +123,11 @@ protected function handleGet( $lastId = 0 ) {
 			$records[$key]['wikiprojects'] = $cleanWikiprojects;
 		}
 		$this->view->set( 'records', $records );
+
+		// determine if we are on the staging environment, so we can show a banner in the view
+		$rootUri = $this->slim->request->getRootUri();
+		$this->view->set( 'staging', preg_match( '/\/plagiabot/', $rootUri ) );
```

Comment:
Let's move this to a middleware instead. This should not be a part of CopyPatrol controller.